<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en_US" lang="en_US">
<head>
<link rel="stylesheet" href="https://twiki.opensciencegrid.org/twiki/pub/TWiki/HeadlinesPlugin/style.css" type="text/css" media="all" />
<title> ResearchHighlight21 &lt; Management &lt; TWiki    </title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
<link rel="icon" href="/twiki/pub/Management/WebPreferences/favicon.ico    " type="image/x-icon" />
<link rel="shortcut icon" href="/twiki/pub/Management/WebPreferences/favicon.ico    " type="image/x-icon" />
<link rel="alternate" href="https://twiki.opensciencegrid.org/bin/edit/Management/ResearchHighlight21?_T=16 Feb 2017" type="application/x-wiki" title="edit ResearchHighlight21" />
<meta name="SCRIPTURLPATH" content="/bin" />
<meta name="SCRIPTSUFFIX" content="" />
<meta name="TEXT_JUMP" content="Jump" />
<meta name="TEXT_SEARCH" content="Search" />
<meta name="TEXT_NUM_TOPICS" content="Number of topics:" />
<meta name="TEXT_MODIFY_SEARCH" content="Modify search" />
<meta name="robots" content="noindex" /><link rel="alternate" type="application/rss+xml" title="RSS Feed" href="/bin/view/Management/WebRss" />    
<base href="https://twiki.opensciencegrid.org/bin/view/Management/ResearchHighlight21"></base>
<!--BEHAVIOURCONTRIB--><script type="text/javascript" src="/twiki/pub/TWiki/BehaviourContrib/behaviour.compressed.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikilib.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiWindow.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiEvent.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiHTML.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiCSS.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiForm.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/PatternSkin/pattern.js"></script><style type="text/css" media="all">
@import url('/twiki/pub/TWiki/TWikiTemplates/base.css');
</style><script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiStyles.js"></script><style type="text/css" media="all">


</style>
<style type="text/css" media="all">
@import url("/twiki/pub/TWiki/TWikiNetSkin/layout.css");
@import url("/twiki/pub/TWiki/TWikiNetSkin/style.css");
@import url("/twiki/pub/TWiki/TWikiNetSkin/colors.css");
@import url("/twiki/pub/TWiki/TWikiNetSkin/rounded_corners.css");
</style>
<style type="text/css" media="all">
	/* Styles that are set using variables */
	#patternLeftBar .patternWebIndicator,
	.patternBookView .twikiTopRow {
		background-color:#D0D0D0;
	}
	.patternBookView {
		border-color:#D0D0D0;
	}
	.patternPreviewPage #patternMain {
		/* uncomment to set the preview image */
		/*background-image:url("/twiki/pub/TWiki/PreviewBackground/preview2bg.gif    ");*/
	}
	
</style><style type="text/css" media="all">



</style>
<style type="text/css" media="all">
	@import url("/twiki/pub/TWiki/TWikiNetSkin/print.css");
</style><!--GOOGLEANALYTICSPLUGIN--><!-- Google Analytics script -->
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-69012-21']);
  _gaq.push(['_setDomainName', 'none']);
  _gaq.push(['_setAllowLinker', true]);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
</head>
<body class="patternViewPage patternPrintPage">
<a name="PageTop"></a>
<div id="patternScreen">
<div id="patternPageShadow">
<div id="patternPage">
<div id="patternOuter">
<div id="patternFloatWrap">
<div id="patternMain">
<div id="patternMainContents">
<div class="patternContent"><div class="patternTopic"> 
<h1><a name="Grid_enabled_virus_hunting"></a> Grid-enabled virus hunting </h1>
<p />
<p />
<p />
DNA sequencing and sequence analysis happens daily in many biological sciences laboratories, but analyzing large sets of genetic data increasingly requires computing resources beyond the capabilities of most labs.
<p />
The search for the best hardware and software led Eric Delwart, a professor of laboratory medicine at the University of California, San Francisco, and a senior investigator at the Blood Systems Research Institute, and Chunlin Wang, a research associate at the Stanford University Genome Technology Center, to the Renaissance Computing Institute's (RENCI) Trash/Engagement Team and then to the distributed computing resources of the <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> and the Open Science Grid (OSG).
<p />
Delwart works with Wang to identify new viruses. The team uses a technique called massively parallel pyrosequencing, which can determine sequences for millions of DNA fragments using high-throughput computing. The resulting DNA sequences are then compared to all the sequences in public sequence databases to identify viral fingerprint sequences. One single sequencing reaction generates massive volumes of data that can take months, even years, to analyze on a small-scale computing cluster.
<p />
In an effort to analyze more data more efficiently, Delwart and Wang turned to the RENCI Trash/Engagement Team, which participates in the <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> Science Gateways program and leads the OSG Trash/Engagement program. <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span>’s Science Gateways aim to bring new communities of users to <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> resources by providing easy access to the <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span>’s distributed computing resources. The OSG Trash/Engagement program recruits new users from a wide range of disciplines and helps them become users of the distributed computing systems operated and maintained by OSG members.
<p />
The effort to find more computing power paid off for the Delwart and Wang: In early May, they used 70,000 CPU hours on <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> and OSG resources to complete in a week a DNA sequence analysis that would have taken over three months on their own lab cluster. The team submitted its jobs to the national resources using a RENCI-developed, Web services-based computational science platform (link to PDF).
Distribution of BLAST jobs from one Pyrosequencing run
<p />
<p />
“We created an application that communicates with RENCI's <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> Science Gateway,” said Jason Reilly, a RENCI senior research software developer. “For the user, it’s very simple — just log in and the application maps the data input to specific tasks to be done. The beauty is you don’t have to submit commands over and over again. You can run hundreds or even thousands of operations and you only have to submit the command once.”
<p />
The custom application created by Reilly was dubbed BLASTMaster because it builds on the Basic Local Alignment Search Tool (BLAST) used to search sequence databases. BLASTMaster divides commands into tasks and pushes the work to RENCI's <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> Science Gateway, which submits, monitors, and manages the compute workload on systems that are part of <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span>’s nationwide network of high performance machines, and to OSG machines. After entering the initial commands, the researchers merely had to wait for their results.
<p />
“Large computer farms that we might use are often composed of heterogeneous smaller clusters,” said Wang. “The BLASTMaster tool and a Web services environment is particularly useful to those of us without much experience using compute clusters. It gives us a uniform interface to submit jobs, which greatly enhances our productiveness.”
<p />
The sequence analysis work used <span class="twikiNewLink">TeraGrid<a href="/bin/edit/Management/TeraGrid?topicparent=Management.ResearchHighlight21" rel="nofollow" title="TeraGrid (this topic does not yet exist; you can create it)">?</a></span> resources at Purdue University (West Layfayette, IN), OSG resources at RENCI (Chapel Hill, NC) and a cluster in the University of North Carolina at Chapel Hill computer science department supported by the National Institutes of Health. The work has real-world value taken straight from recent headlines about the <span class="twikiNewLink">H1N1<a href="/bin/edit/Management/H1N1?topicparent=Management.ResearchHighlight21" rel="nofollow" title="H1N1 (this topic does not yet exist; you can create it)">?</a></span> virus.
<p />
“Knowing the genomic sequence of a human virus allows for quicker diagnostics to identify infections,” said Delwart. “Quicker diagnostics can lead to more informed decisions on how an emerging virus is spread and how to control it. Knowing the sequence can also help make vaccines or anti-virals against that virus.”
<p />
~ Karen Green, RENCI
June 2009
<p />
This item also appeared in iSGTW.  
<p />
<p />
-- <a href="/bin/view/Main/RuthPordes" class="twikiLink">RuthPordes</a> - 22 Jan 2012</div><!-- /patternTopic-->
<p />
<p />
</div><!-- /patternContent-->
<hr />
This topic: Management<span class='twikiSeparator'>&nbsp;&gt;&nbsp;</span><a href="/bin/view/Management/OSGResearchHighlights" class="twikiLink">OSGResearchHighlights</a><span class='twikiSeparator'>&nbsp;&gt;&nbsp;</span>ResearchHighlight21</span> <br />    
Topic revision: r2 - 06 May 2016 - 19:24:14 - <a href="/bin/view/Main/KyleGross" class="twikiLink">KyleGross</a>
</div><!-- /patternMainContents-->
</div><!-- /patternMain-->
</div><!-- /patternFloatWrap-->
<div class="clear">&nbsp;</div>
</div><!-- /patternOuter--><div id="patternBottomBar"><div id="patternBottomBarContents"><div id="twikinetBadge"><a href="http://www.twiki.net/"><img src="https://twiki.opensciencegrid.org/twiki/pub/TWiki/TWikiNetSkin/twiki-badge-88x31.gif" alt="TWIKI.NET" width="88" height="31" border="0" /></a></div><!--/twikinetBadge--><div id="patternWebBottomBar"><p>
<font size="-1">
TWiki |
<a href="https://ticket.grid.iu.edu/goc/twiki">Report Bugs</a> |
<a href="https://twiki.grid.iu.edu/bin/view/Operations/IUPrivacyPolicy">Privacy Policy</a>
</p>
<p>
<font size="-2">
<span class="twikiRight"> <a href="http://twiki.org/"><img src="/twiki/pub/TWiki/TWikiLogos/T-logo-80x15.gif" alt="This site is powered by the TWiki collaboration platform" width="80" height="15" title="This site is powered by the TWiki collaboration platform" border="0" /></a></span>Copyright by the contributing authors. All material on this collaboration platform is the property of the contributing authors..
</font>
</p></div><!--/patternWebBottomBar--></div><!-- /patternBottomBarContents--></div><!-- /patternBottomBar-->
</div><!-- /patternPage-->
</div><!-- /patternPageShadow-->
</div><!-- /patternScreen-->
</body></html>
<p />