<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en_US" lang="en_US">
<head>
<link rel="stylesheet" href="https://twiki.opensciencegrid.org/twiki/pub/TWiki/HeadlinesPlugin/style.css" type="text/css" media="all" />
<title> UserSchool15Wed12precompiled &lt; Education &lt; TWiki    </title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
<link rel="icon" href="/twiki/pub/Education/WebPreferences/favicon.ico    " type="image/x-icon" />
<link rel="shortcut icon" href="/twiki/pub/Education/WebPreferences/favicon.ico    " type="image/x-icon" />
<link rel="alternate" href="https://twiki.opensciencegrid.org/bin/edit/Education/UserSchool15Wed12precompiled?_T=16 Feb 2017" type="application/x-wiki" title="edit UserSchool15Wed12precompiled" />
<meta name="SCRIPTURLPATH" content="/bin" />
<meta name="SCRIPTSUFFIX" content="" />
<meta name="TEXT_JUMP" content="Jump" />
<meta name="TEXT_SEARCH" content="Search" />
<meta name="TEXT_NUM_TOPICS" content="Number of topics:" />
<meta name="TEXT_MODIFY_SEARCH" content="Modify search" />
<meta name="robots" content="noindex" /><link rel="alternate" type="application/rss+xml" title="RSS Feed" href="/bin/view/Education/WebRss" />    
<base href="https://twiki.opensciencegrid.org/bin/view/Education/UserSchool15Wed12precompiled"></base>
<!--BEHAVIOURCONTRIB--><script type="text/javascript" src="/twiki/pub/TWiki/BehaviourContrib/behaviour.compressed.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikilib.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiWindow.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiEvent.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiHTML.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiCSS.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiForm.js"></script>
<script type="text/javascript" src="/twiki/pub/TWiki/PatternSkin/pattern.js"></script><style type="text/css" media="all">
@import url('/twiki/pub/TWiki/TWikiTemplates/base.css');
</style><script type="text/javascript" src="/twiki/pub/TWiki/TWikiJavascripts/twikiStyles.js"></script><style type="text/css" media="all">


</style>
<style type="text/css" media="all">
@import url("/twiki/pub/TWiki/TWikiNetSkin/layout.css");
@import url("/twiki/pub/TWiki/TWikiNetSkin/style.css");
@import url("/twiki/pub/TWiki/TWikiNetSkin/colors.css");
@import url("/twiki/pub/TWiki/TWikiNetSkin/rounded_corners.css");
</style>
<style type="text/css" media="all">
	/* Styles that are set using variables */
	#patternLeftBar .patternWebIndicator,
	.patternBookView .twikiTopRow {
		background-color:#D0D0D0;
	}
	.patternBookView {
		border-color:#D0D0D0;
	}
	.patternPreviewPage #patternMain {
		/* uncomment to set the preview image */
		/*background-image:url("/twiki/pub/TWiki/PreviewBackground/preview2bg.gif    ");*/
	}
	
</style><style type="text/css" media="all">

@import url('/pub/Documentation/Tools/exercises.css ');

</style>
<style type="text/css" media="all">
	@import url("/twiki/pub/TWiki/TWikiNetSkin/print.css");
</style><!--GOOGLEANALYTICSPLUGIN--><!-- Google Analytics script -->
<script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-69012-21']);
  _gaq.push(['_setDomainName', 'none']);
  _gaq.push(['_setAllowLinker', true]);
  _gaq.push(['_trackPageview']);
  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>
</head>
<body class="patternViewPage patternPrintPage">
<a name="PageTop"></a>
<div id="patternScreen">
<div id="patternPageShadow">
<div id="patternPage">
<div id="patternOuter">
<div id="patternFloatWrap">
<div id="patternMain">
<div id="patternMainContents">
<div class="patternContent"><div class="patternTopic"> <style type="text/css">
pre em { font-style: normal; background-color: yellow; }
pre strong { font-style: normal; font-weight: bold; color: #008; }
</style>
<p />
<h1><a name="Wednesday_Exercise_1_2_Using_a_P"></a> Wednesday Exercise 1.2: Using a Pre-compiled Binary </h1>
<p />
In this exercise, we will run a job using a downloaded, pre-compiled binary.  This exercise should take 10-15 minutes.  
<p />
<h2 class="twikinetRoundedAttachments"><span class="twikinetHeader"><a name="Background"></a> Background </span></h2>
<p />
In the previous exercise, we used a piece of software that was available as source code, and 
could be compiled to a static binary.  However, software is available in many different forms.  
You may also encounter scientific software provided as a static binary.  Here, the software provider 
has compiled the software for you (typically on several operating systems).  
Using a pre-compiled binary means you can avoid compiling the code yourself; 
accessing the software and getting it ready to run is as simple as downloading the binary.  
<p />
<h2 class="twikinetRoundedAttachments"><span class="twikinetHeader"><a name="Our_Software_Example"></a> Our Software Example </span></h2>
<p />
The software we will be using for this example is a common tool for aligning 
genome and protein sequences against a reference database, the BLAST program.  
<p /> <ol>
<li> Search the internet for the BLAST software.  Searches might include "blast executable or "download blast software".  Hopefully these searches will lead you to a BLAST website page that looks like this:    <p> <img alt="blast_executable_page.png" src="https://twiki.opensciencegrid.org/twiki/pub/Education/UserSchool15Wed12precompiled/blast_executable_page.png" /> </p>
</li> <li> Find the link to the downloads page for BLAST software.  
</li> <li> We could download the source and compile it ourselves, but instead, we're going to use one of the pre-built binaries.  Before proceeding, look at the list of downloads and try to determine which one you want.  
</li> <li> Based on our operating system, we want to use the Linux binary, which is labelled with the <code>x64-linux</code> suffix.  All the other links are either for source code or other operating systems.  While logged into <code>osg-ss-glidein-submit.chtc.wisc.edu</code>, create a directory for this exercise.  Then download the appropriate <code>tar.gz</code> file and un-tar it.  <pre class="screen">
$> <strong>wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/ncbi-blast-2.2.31+-x64-linux.tar.gz</strong>
$> <strong>tar xzf ncbi-blast-2.2.31+-x64-linux.tar.gz</strong>
</pre>
</li> <li> We're going to be using the <code>blastx</code> binary in our job.  Where is it in the directory you just downloaded?  
</li></ol> 
<p />
<h2 class="twikinetRoundedAttachments"><span class="twikinetHeader"><a name="Copy_the_Input_Files"></a> Copy the Input Files </span></h2>
<p />
To run BLAST, we need an input file and reference database.  For 
this example, we'll use the "pdbaa" database, which contains sequences 
for the protein structure from the Protein Data Bank.  For our input file, we'll use an abbreviated 
fasta file with mouse genome information.  These files are located on the 
OSG submit node at <code>/home/ckoch5/OSGdata</code>.  
<p /> <ol>
<li> Copy them to your current directory: <pre class="screen">
$> <strong>cp -r /home/ckoch5/OSGdata/mouse.fa /home/ckoch5/OSGdata/pdbaa ./</strong>
</pre>
</li></ol> 
<p />
<h2 class="twikinetRoundedAttachments"><span class="twikinetHeader"><a name="Submitting_the_Job"></a> Submitting the Job </span></h2>
<p />
We now have our program (the pre-compiled <code>blastx</code> binary) and our input files, so all that 
remains is to create the submit file.  A typical <code>blastx</code> command looks something like this: 
<p />
<pre class="screen">
$> <strong> blastx -db database -query input_file -out results.txt</strong>
</pre>
<p /> <ol>
<li>  Copy the submit file from the last exercise into your current directory.  
</li> <li> Think about which lines you will need to change or add to your submit file in order to submit the job successfully. In particular:  <ul>
<li> What is the executable?
</li> <li> How can you indicate the entire command line sequence above?  
</li> <li> Which files need to be transferred in addition to the executable?  
</li> <li> Does this job require a certain type of operating system?  
</li></ul> 
</li> <li> Try to answer these questions and modify your submit file appropriately.  
</li> <li> Once you have done all you can, check your submit file against the lines below, which contain the necessary changes to run this particular job.   <ul>
<li> The executable is <code>blastx</code>, which is located in the <code>bin</code> directory of our downloaded BLAST directory.  We need to use the <code>arguments</code> line in the submit file to express the rest of the command. <pre class="file">
executable = ncbi-blast-2.2.31+/bin/blastx
arguments = "-db pdbaa -query mouse.fa -out results.txt"
</pre>
</li> <li> The BLAST program requires our input file and database, so they must be transferred with <code>transfer_input_files</code>. <pre class="file">
transfer_input_files = pdbaa/, mouse.fa
</pre>*Note* : It is important to include the <code>/</code> after <code>pdbaa</code>.  Including the <code>/</code> means that Condor will transfer the database files from <code>pdbaa/</code> directly into the job's working directory, which is what we want for this example. 
</li> <li> Because we downloaded a Linux-specific binary, we need to request machines that are running Linux.  <pre class="file">
requirements = (OpSys == "LINUX")
</pre>
</li></ul> 
</li> <li> Submit the blast job using <code>condor_submit</code>.  Once the jobs starts, it should run in just a few minutes and produce a file called <code>results.txt</code>. 
</li></ol> </div><!-- /patternTopic-->
<p />
<p />
</div><!-- /patternContent-->
<hr />
This topic: Education<span class='twikiSeparator'>&nbsp;&gt;&nbsp;</span><a href="/bin/view/Education/OSGUserSchool2015" class="twikiLink">OSGUserSchool2015</a> &gt; <a href="/bin/view/Education/UserSchool15Materials" class="twikiLink">UserSchool15Materials</a><span class='twikiSeparator'>&nbsp;&gt;&nbsp;</span>UserSchool15Wed12precompiled</span> <br />    
Topic revision: r13 - 28 Jul 2015 - 13:35:11 - <a href="/bin/view/Main/ChristinaKoch" class="twikiLink">ChristinaKoch</a>
</div><!-- /patternMainContents-->
</div><!-- /patternMain-->
</div><!-- /patternFloatWrap-->
<div class="clear">&nbsp;</div>
</div><!-- /patternOuter--><div id="patternBottomBar"><div id="patternBottomBarContents"><div id="twikinetBadge"><a href="http://www.twiki.net/"><img src="https://twiki.opensciencegrid.org/twiki/pub/TWiki/TWikiNetSkin/twiki-badge-88x31.gif" alt="TWIKI.NET" width="88" height="31" border="0" /></a></div><!--/twikinetBadge--><div id="patternWebBottomBar"><p>
<font size="-1">
TWiki |
<a href="https://ticket.grid.iu.edu/goc/twiki">Report Bugs</a> |
<a href="https://twiki.grid.iu.edu/bin/view/Operations/IUPrivacyPolicy">Privacy Policy</a>
</p>
<p>
<font size="-2">
<span class="twikiRight"> <a href="http://twiki.org/"><img src="/twiki/pub/TWiki/TWikiLogos/T-logo-80x15.gif" alt="This site is powered by the TWiki collaboration platform" width="80" height="15" title="This site is powered by the TWiki collaboration platform" border="0" /></a></span>Copyright by the contributing authors. All material on this collaboration platform is the property of the contributing authors..
</font>
</p></div><!--/patternWebBottomBar--></div><!-- /patternBottomBarContents--></div><!-- /patternBottomBar-->
</div><!-- /patternPage-->
</div><!-- /patternPageShadow-->
</div><!-- /patternScreen-->
</body></html>
<p />